### ISSUE TITLE
[Figma] Button sizing/padding issues when added from shared lib

### ISSUE THREAD

**Issue by uncleserb** [carbon-design-system/carbon-design-kit#694]
## Detailed description

I have this very odd behavior with the Button component. When added to a Figma project (having set Carbon v11 to a shared library), the Button is rendered with a broken padding-right/sizing issue.

Relevant screen grab:

https://github.com/carbon-design-system/carbon-design-kit/assets/1575503/e9c20988-c96d-4c3a-b750-63825dc4131e

When I check the source (Carbon's v11 shared library file) the button component seems to render fine but as soon as I add it to a project it renders like that.
This strange behavior began a couple of months ago. I therefore checked the Carbon v11 shared library in Figma to view its version history (maybe I or someone else had simply tampered with the file unknowingly) but alas, even after restoring the orginal version, this odd rendering has been persisting.

Another odd behavior that I believe may be related: this started when viewing the Button in Prototype mode, we noticed this resizing of the component before our eyes.

Relevant screen grabs:

https://github.com/carbon-design-system/carbon-design-kit/assets/1575503/60e2ff88-5185-4171-a87a-50c84ca2c096

https://github.com/carbon-design-system/carbon-design-kit/assets/1575503/2418b19c-f461-4f90-9df8-17e2c2c73cd0

As you can see, the component seems to render fine but as soon as it is hoevered, the padding/sizing glitches and it seems impossible to get back to the original state.

> What version of the Carbon Design System Kit are you using?

Figma v11 (White theme)

> Is this issue related to a specific component? If so please give the page, artboard, and layer/symbol name.

The Button component (and therefore all instances where it is slotted such as Table headers, ...).

> What did you expect to happen? What happened instead? What would you like to see changed?

To be consistent with the core design system library/guidelines on buttons.

> Is this issue being filed based on a discrepancy between the website/code and the kit? If so what is the inconsistency?

No, the kit looks fine (see attached screen grab) when viewed on its own. The issue occurs when adding said component to a Figma project file (see other attached screen grab).

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

Nothing internal to IBM, using Carbon for a software project at a private company.

## Additional information

![carbon-figma-glitchy-render](https://github.com/carbon-design-system/carbon-design-kit/assets/1575503/4cbd784f-0ed4-4b37-8a00-1d78d1e9e054)
![carbon-figma-shared-lib](https://github.com/carbon-design-system/carbon-design-kit/assets/1575503/d2c3c2f7-9f22-46c4-95fd-d94baf694933)

## Add labels

figma, bug, button

**Comment by uncleserb** [comment]
@laurenmrice if needed; I can provide access to my project(s) if you need assistance to reproduce this

**Comment by laurenmrice** [comment]
Hey @uncleserb 👋

Are you still experiencing this issue? I am unable to reproduce on my end. I could take a look at your project to try to reproduce it that way its you are still experiencing the same issues.

**Comment by uncleserb** [comment]
Hey @laurenmrice , thanks for getting back to me on this. I was able to workaround the issue by updating the _Auto-Layout_ spacing value for the Button components by switching the value from `Auto` to `32px` (the default value from Carbon guidelines).
This fixes the sizing issue I had been experiencing even though not ideal. I have a feeling this issue may have been caused by Figma's update to _Auto-Layout_ a couple of months ago.
If you weren't able to reproduce the problem, don't hassle any further! Many thanks for your feedback 🙏
